packages/mui-material/src/Chip/Chip.js

```diff
@@ -43,6 +44,18 @@ const useUtilityClasses = (ownerState) => {
   return composeClasses(slots, getChipUtilityClass, classes);
 };
 
+const parseIconColor = (icon, color) =>
+  React.isValidElement(icon) ? icon.props.color || color : color;
+
+const isIconColorEqualToColor = ({ icon, color }) => {
+  const iconColor = parseIconColor(icon, color);
+
+  return iconColor === color;
+};
+
+const isClickable = ({ clickable, onClick }) =>
+  clickable !== false && !!onClick ? true : clickable;
+
```

Comment:
It might be possible to remove this if we implement the related styles with https://github.com/mui/material-ui/tree/master/packages/pigment-css-react#styling-based-on-runtime-values. Would that be better?